provisio-model/src/main/java/ca/vanzyl/provisio/model/ProvisioArtifact.java

```diff
@@ -146,7 +146,7 @@ public File getFile() {
     }
 
     @Override
-    public Artifact setFile(File file) {
+    public ProvisioArtifact setFile(File file) {
         delegate = delegate.setFile(file);
         return this;
```

Comment:
Doesn't that break the assumption of the resolver that the Artifact is immutable ?

Reply:
Yes, it does, will fix ProvisioArtifact as well

Reply:
Fixed